src/services/trackingPlan.ts

```diff
@@ -73,7 +73,7 @@ export class TrackingPlanservice {
 
     // capture overall function latency
     // with metadata tags
-    stats.histogram('tp_batch_validation_latency', startTime, {
+    stats.histogram('tp_batch_validation_latency', (Date.now() - startTime) / 1000, {
```

Comment:
Diving by 1000 to report metric in seconds instead of ms as the metric for the `timing` is also seconds.